lib/matplotlib/tests/test_patches.py

```diff
@@ -149,6 +149,40 @@ def test_rotate_rect_draw(fig_test, fig_ref):
     assert rect_test.get_angle() == angle
 
 
+@check_figures_equal(extensions=['png'])
+def test_dash_offset_patch_draw(fig_test, fig_ref):
+    ax_test = fig_test.add_subplot()
+    ax_ref = fig_ref.add_subplot()
+
+    loc = (0, 0)
+    width, height = (1, 1)
+    edgecolor = 'b'
+    linestyle = (0, [6, 6])
+    linestyle_hacked = (0, [0, 6, 6, 0])
+    rect_ref = Rectangle(loc, width, height, edgecolor=edgecolor,
+                                                linestyle=linestyle)
+    rect_ref2 = Rectangle(loc, width, height, edgecolor=edgecolor,
+                                        linestyle=linestyle_hacked)
```

Comment:
I first thought: Why do we need two reference rectangles at all? We have two test rectangles that are carefully crafted  to lie on top of each other, but have shifted dashes. Can't we have one single solid rectangle as reference?

Apparently the two dashed lines don't add up to a fully solid line because of antialiasing.

If we can't get a solid rectangle anyway, let's not even try: I propose to leave the rectangles on top of each other, but use different colors.

```suggestion
    rect_ref = Rectangle(loc, width, height, linewdith=3,
                         edgecolor='b','linestyle=(0, [6, 6]))
    # fill the line gaps using a linestyle (0, [0, 6, 6, 0]), which is
    # equivalent to (6, [6, 6]) but has 0 dash offset
    rect_ref2 = Rectangle(loc, width, height, linewidth=3,
                          edgecolor='r', linestyle=(0, [0, 6, 6, 0]))
```
Note that I have inlined the edgecolor and linestyle variables. They are only used once and IMHO defining them explicitly leads to another indirection and additional lines, which don't help readability. I've also increased the `linewidth` to 3, which makes the dash pattern better visible.

Reply:
Thanks for the comments on code readability. It's really useful.